### PR TITLE
Add more analytics events for limit orders

### PR DESCRIPTION
**Pull request by nenadV91**
# Summary

This PR adds following events to limit orders
```
Send
Sign
SignAndSend
SignToSelf
Reject
Error
```

# To test
- install [GA debug](https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna?hl=en) and turn it on
- open dev-tools and go to console
- go to limit orders page
- try to create an order and reject/approve signing
- you should see these events in console

### CODE REVIEW

**Comment by elena-zh** [comment]
Hey @nenadV91 , I think that `SignAndSend` and `SignToSelf` events are mixed up: 
 - I see SignToSelf when I send an order to Recipient
 - SignAndSend is displayed when I place an order without Recipient filled

mixed up

Could you please also clarify how can I get 'Sigh' event? 
When I place an order I see only `Send Limit Order`, `SignAndSend` , `SignToSelf`, `Posted Limit Order` events. 

Thanks!

**Comment by elena-zh** [comment]
One more issue: when I reject signing order in Gnosis safe, I see 'Error' event instead of the 'Reject'
image

File: src/cow-react/modules/limitOrders/services/tradeFlow/index.ts [hunk 1]
```diff
@@ -11,6 +11,11 @@ import { Web3Provider } from '@ethersproject/providers'
 import { AddAppDataToUploadQueueParams, AppDataInfo } from 'state/appData/types'
 import confirmPriceImpactWithoutFee from '@src/components/swap/confirmPriceImpactWithoutFee'
 import { LOW_RATE_THRESHOLD_PERCENT } from '@cow/modules/limitOrders/const/trade'
+import { swapFlowAnalytics } from '@cow/modules/swap/services/common/steps/analytics'
+import { logSwapFlow } from '@cow/modules/swap/services/utils/logger'
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/services/tradeFlow/index.ts:15]: We mustn't use files from swap module in the limit-orders module.
If something is common, then it must be located in the `common` or `trade` module

Comment (anxolin) [reply src/cow-react/modules/limitOrders/services/tradeFlow/index.ts:15]: I wonder if this is sth that eslint could help us with, define our own rules. Maybe like this we can reduce progressively the coupling of modules

File: src/cow-react/modules/limitOrders/services/tradeFlow/index.ts [hunk 2]
```diff
@@ -34,41 +39,68 @@ export async function tradeFlow(
...
+  const marketLabel = [sellToken.symbol, buyToken.symbol].join(',')
+  const swapFlowAnalyticsContext: SwapFlowAnalyticsContext = {
+    account,
+    recipient,
+    recipientAddress: recipientAddressOrName,
+    marketLabel,
+    orderClass: OrderClass.LIMIT,
+  }
+
+  logSwapFlow('LIMIT ORDER FLOW', 'STEP 1: confirm price impact')
```
Comment (shoom3301) [inline src/cow-react/modules/limitOrders/services/tradeFlow/index.ts:53]: Limit orders module should have its own logger

File: src/cow-react/modules/limitOrders/services/tradeFlow/index.ts (hunk 1 above)
Comment (shoom3301) [reply src/cow-react/modules/limitOrders/services/tradeFlow/index.ts:15]: I haven't heard about eslint rules for that, but would be great to control it somehow

**Comment by nenadV91** [comment]
@shoom3301 I think we can refactor this in the future.
For now I would just like to have these events working in production so people who need analytics can see them.

**Comment by shoom3301** [comment]
@nenadV91 added small refactoring, to align links between modules:
https://github.com/cowprotocol/cowswap/pull/1729

**Comment by elena-zh** [comment]
@nenadV91 , just 1 pending question: 

> Could you please also clarify how can I get 'Sigh' event?
> When I place an order I see only Send Limit Order, SignAndSend , SignToSelf, Posted Limit Order events.

Thanks

**Comment by nenadV91** [comment]
`Sign` is when there is no recipient defined.

**Comment by elena-zh** [comment]
> @nenadV91 , just 1 pending question:
> 
> > Could you please also clarify how can I get 'Sigh' event?
> > When I place an order I see only Send Limit Order, SignAndSend , SignToSelf, Posted Limit Order events.
> 
> Thanks

Nevermind, I checked the difference while testing the current changes. Everything LGTM now. Thank you!

**Review by elena-zh — APPROVED** [review]

**Review by elena-zh — COMMENTED** [review]
UPD
@nenadV91 , I noticed, that  BE error pop-ups are no longer displayed in this PR:
image

Compare with DEV:
image

**Comment by nenadV91** [comment]
@elena-zh I think its fixed now

**Comment by elena-zh** [comment]
Looks good now, thank you!

**Comment by nenadV91** [comment]
Merging this and we can do a post code review

**Review by alfetopito — APPROVED** [review]

**Pull request merged**